packages/schema-generator/src/generate_schema.test.ts

```diff
@@ -54,22 +54,60 @@ void describe('SchemaGenerator', () => {
     assert.strictEqual(dbConfig.port, 3306);
   });
 
-  void it('should throw error if one or more parts are missing in the database url', async () => {
-    const parse = () => parseDatabaseUrl('mysql://hostname/db');
-    assert.throws(parse, {
-      name: 'DatabaseUrlParseError',
-      message:
-        'Unable to parse the database URL. One or more parts of the database URL is missing. Missing [username, password, port].',
-    });
+  void it('should parse postgres url correctly', async () => {
+    const dbConfig = parseDatabaseUrl(
+      'postgresql://user:password@test-host-name:5432/db'
+    );
+    assert.strictEqual(dbConfig.engine, 'postgresql');
+    assert.strictEqual(dbConfig.username, 'user');
+    assert.strictEqual(dbConfig.password, 'password');
+    assert.strictEqual(dbConfig.host, 'test-host-name');
+    assert.strictEqual(dbConfig.database, 'db');
+    assert.strictEqual(dbConfig.port, 5432);
+  });
+
+  void it('should parse database url with postgres protocol correctly', async () => {
+    const dbConfig = parseDatabaseUrl(
+      'postgres://user:password@test-host-name:5432/db'
+    );
+    assert.strictEqual(dbConfig.engine, 'postgresql');
+    assert.strictEqual(dbConfig.username, 'user');
+    assert.strictEqual(dbConfig.password, 'password');
+    assert.strictEqual(dbConfig.host, 'test-host-name');
+    assert.strictEqual(dbConfig.database, 'db');
+    assert.strictEqual(dbConfig.port, 5432);
   });
 
-  void it('should throw error if the connection uri is missing port', async () => {
```

Comment:
This is not a valid case anymore. If port number is not present, we will use the default port number.